### ISSUE TITLE
Can we run test_on_single_video.py with 3D pretrained model?

### ISSUE THREAD

**Issue by Marcovaldong** [facebookresearch/DetectAndTrack#29]
I tested the test_on_single_video.py with 3D pretrained model by running following command:
```
python tools/test_on_single_video.py --cfg configs/video/3d/03_R-18-3D_PTFromCOCO.yaml --video ~/shiqi.dong/try/ava/P90hF2S1JzA.avi --output ./visualization/P90/ TEST.WEIGHTS pretrained_models/configs/video/3d/03_R-18-3D_PTFromCOCO.yaml_dir/model_final.pkl
```
But got following error:
```python
......
INFO net.py: 249: fc7_b preserved in workspace (unused)
I0807 10:22:48.503657  9865 operator.cc:169] Engine CUDNN is not available for operator MaxPool.
I0807 10:22:48.505997  9865 net_dag_utils.cc:102] Operator graph pruning prior to chain compute took: 5.64e-05 secs
I0807 10:22:48.506840  9865 operator.cc:169] Engine CUDNN is not available for operator MaxPool.
I0807 10:22:48.507918  9865 net_dag_utils.cc:102] Operator graph pruning prior to chain compute took: 2.4698e-05 secs
I0807 10:22:48.509433  9865 net_dag_utils.cc:102] Operator graph pruning prior to chain compute took: 1.8558e-05 secs
Processing Detection for Frame 1
Traceback (most recent call last):
  File "tools/test_on_single_video.py", line 340, in <module>
    main(name_scope, gpu_dev, num_images, args)
  File "tools/test_on_single_video.py", line 264, in main
    model, im_, None)                                        #TODO: Parallelize detection
  File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/core/test.py", line 905, in im_detect_all
    scores, boxes, im_scales = im_detect_bbox(model, im, box_proposals)
  File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/core/test.py", line 174, in im_detect_bbox
    inputs, im_scales = _get_blobs(im, boxes)
  File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/core/test.py", line 147, in _get_blobs
    blobs['data'], im_scale_factors = _get_image_blob(im)
  File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/core/test.py", line 73, in _get_image_blob
    blob = blob_utils.im_list_to_blob(processed_ims)
  File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/utils/blob.py", line 66, in im_list_to_blob
    blob = image_utils.move_batch_to_time(blob, cfg.VIDEO.NUM_FRAMES)
  File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/utils/image.py", line 85, in move_batch_to_time
    assert(blob.shape[0] % T == 0)
AssertionError
```
So can we run test_on_single_video.py with 3D pretrained model?@yannadani

**Comment by rohitgirdhar** [comment]
Doesn't look like, but would be good to have. One thing to fix would be to read multiple frames [here](https://github.com/facebookresearch/DetectAndTrack/blob/0b612c126255d74cb346943cf1ad4d8157b68509/tools/test_on_single_video.py#L259).

**Comment by yanxiangyi** [comment]
Same problem.

**Comment by Marcovaldong** [comment]
I finished the change, now I can run test_on_single_video.py  with 3D pretrained model (e.g. 03_R-18-3D_PTFromCOCO-4GPU-BATCH1.yaml). While the result looks worse than the result got through 2D pretrained model. Is it because that the 3D pretrained model's backbone network is ResNet18, while the 2D pretrained model is based on ResNet101? Can we get best 3D models based on ResNet101? @rohitgirdhar

**Comment by heury** [comment]
How can you solve AssertionError? I have a same issue @Marcovaldong

**Comment by Marcovaldong** [comment]
@heury I got the error when I try to run test_on_single_video.py when using 3D pretrained model. I added several code in the script, then fix it and run test_on_single_video.py with 3D pretrained model successfully.

I don't know how you got error, please show us your detail.

**Comment by rohitgirdhar** [comment]
The 3D model is worse than the best 2D model due to smaller frames, shallow network etc. It should be better than the comparable 2D model though.

**Comment by heury** [comment]
@Marcovaldong 
I got exact same error you wrote. I also tried test_on_single_video.py with 3D pretrained model.

File "/data/zairan.wang/shiqi.dong/DetectAndTrack/lib/utils/image.py", line 85, in move_batch_to_time
    assert(blob.shape[0] % T == 0)

Can you share your modified code?

**Comment by Marcovaldong** [comment]
@heury OK, I will open a pull request this weekend.

**Comment by rohitgirdhar** [comment]
Resolved by #39
